src/chips/sam4l/dma.rs

```diff
@@ -212,6 +216,25 @@ impl DMAChannel {
 
         volatile_store(&mut registers.interrupt_enable, 1 << 1);
     }
+
+    pub fn do_xfer_buf(&self, pid: usize, 
+                       buf: Option<&'static mut [u8]>,
+                       len: usize) {
+        let ptr = buf.unwrap();
+        if len > ptr.len() {
```

Comment:
OK, totally fine for now then (i'll add the `min` back in). But both this and the redundancy with `do_xfer` is begging for a type. Especially since passing buffers around is a pretty basic task that will almost certainly be used in similar ways on every chip.